### PR TITLE
chore: add Windows and MacOS CI

### PR DESCRIPTION
**Pull request by gregmagolan**
Windows & MacOS CI run on main branch only and only for `e2e/bzlmod` job for now. They will run on PRs if the branch name contains `windows` or `macos` respectively.

This PR also removes RBE CI from PRs to reduce the matrix size for PRs.

---

### Type of change

- Chore (any other change that doesn't affect source or test files, such as configuration)

### Test plan

This PR will run the new CI matrix jobs

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -234,8 +283,18 @@ jobs:
...
-              # Writes EngFlow credential files for RBE configurations
+            - name: Configure Bazel version (Windows)
+              if: matrix.os == 'windows-latest'
+              working-directory: ${{ matrix.folder }}
+              # Overwrite the .bazelversion instead of using USE_BAZEL_VERSION so that Bazelisk
+              # still bootstraps Aspect CLI from configuration in .bazeliskrc. Aspect CLI will
+              # then use .bazelversion to determine which Bazel version to use.
+              run: |
+                  echo "${{ matrix.bazelversion }}" > .bazelversion
+                  $BAZEL_MAJOR_VERSION = "${{ matrix.bazelversion }}".substring(0, 1)
```
Comment (jbedard) [inline .github/workflows/ci.yaml:294]: Where is `BAZEL_MAJOR_VERSION` used? That's not used with non-windows?

Comment (gregmagolan) [reply .github/workflows/ci.yaml:294]: Nice catch. That is a copy-paste left-over from https://github.com/aspect-build/bazel-lib/blob/794df714d7efbf5f2b986470428bea311f4fd772/.github/workflows/ci.yaml#L168 where it is used.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -234,8 +273,16 @@ jobs:
...
 
-            - name: Write EngFlow credentials
-              # Writes EngFlow credential files for RBE configurations
+            - name: Configure Bazel version (Windows)
+              if: matrix.os == 'windows-latest'
+              working-directory: ${{ matrix.folder }}
+              # Overwrite the .bazelversion instead of using USE_BAZEL_VERSION so that Bazelisk
+              # still bootstraps Aspect CLI from configuration in .bazeliskrc. Aspect CLI will
+              # then use .bazelversion to determine which Bazel version to use.
+              run: echo "${{ matrix.bazelversion }}" > .bazelversion
```
Comment (jbedard) [inline .github/workflows/ci.yaml:282]: Why do we have a windows specific "Configure Bazel version"? It seems like we only drop the bazel5 setup in the windows one?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -273,6 +321,18 @@ jobs:
                   ASPECT_NPM_AUTH_TOKEN: ${{ secrets.ASPECT_NPM_AUTH_TOKEN }}
                   ASPECT_GH_PACKAGES_AUTH_TOKEN: ${{ secrets.ASPECT_GH_PACKAGES_AUTH_TOKEN }}
 
+            - name: bazel test //... (Windows)
```
Comment (jbedard) [inline .github/workflows/ci.yaml:324]: This just drops the `ls | ... | grep`? Was that causing issues worth copy+pasting this for windows?

**Review by jbedard — APPROVED** [review]

**Pull request merged**